### PR TITLE
Javascript sdk migration

### PR DESCRIPTION
**Pull request by tomijaga**
Resolves: #585

@angle943 I've made all the necessary migrations using the js sdk

### CODE REVIEW

**Comment by angle943** [comment]
@tomijaga can you resolve the merge conflict?

**Comment by tomijaga** [comment]
@angle943 I did already. I just don't have the rights to merge it.
Here's the PR: https://github.com/thenewboston-developers/Account-Manager/pull/618

**Comment by zinoadidi** [comment]
Conflicts merged @angle943

**Comment by angle943** [comment]
@zinoadidi @tomijaga thanks guys. Please give me about a week to review this. This week I will be very busy moving in to my new home. I will be free starting next week

**Review by sno2 — COMMENTED** [review]
Nice work, here's my comments on a few things.

File: src/renderer/dispatchers/banks.ts [hunk 1]
```diff
@@ -74,18 +76,16 @@ export const fetchBankBankTransactions = (
...
 };
 
 export const fetchBankConfig = (address: string) => async (
   dispatch: AppDispatch,
 ): Promise<{address: string; data?: BankConfig; error?: any}> => {
   try {
-    const {data: rawData} = await axios.get<RawBankConfig>(`${address}/config`, {timeout: AXIOS_TIMEOUT_MS});
-    const data = sanitizePortFieldFromRawBankConfig(rawData);
+    const rawData = await new Bank(address).getConfig();
+    const data = sanitizePortFieldFromRawBankConfig(rawData as RawBankConfig);
```
[the remark this answers is not in the document]
Comment (tomijaga) [reply src/renderer/dispatchers/banks.ts:88]: It's coz the NodeType in the account manager repo is an enum while in the js repo it a type that accepts multiple values ("Bank" | "Primary_Validator" )

[the remark this answers is not in the document]
Comment (tomijaga) [reply src/renderer/dispatchers/banks.ts:88]: Should i just change the `NodeType` in the js repo to an enum?

File: src/renderer/containers/Validator/AddValidatorSigningKeysModal/index.tsx [hunk 2]
```diff
@@ -55,8 +55,8 @@ const AddValidatorSigningKeysModal: FC<ComponentProps> = ({close}) => {
 
   const checkPrivateSigningKey = (publicKey: string, privateKey: string): boolean => {
     try {
-      const {publicKeyHex} = getKeyPairFromSigningKeyHex(privateKey);
-      return publicKeyHex === publicKey;
+      const {accountNumberHex} = new Account(privateKey);
+      return accountNumberHex === publicKey;
     } catch (error) {
       return false;
     }
```
[the remark this answers is not in the document]
Comment (tomijaga) [reply src/renderer/containers/Validator/AddValidatorSigningKeysModal/index.tsx:62]: I made a PR for this at #619

**Comment by zinoadidi** [comment]
@tomijaga some conflicts in package lock. Try to fix locally and make a PR

**Comment by zinoadidi** [comment]
@angle943 still waiting

**Comment by mrbusysky** [comment]
Could likely close this now since the new wallet is being made currently for the beta.

**Comment by zinoadidi** [comment]
Closing in favour of beta

**Pull request closed without merging**